ranker.py

```diff
@@ -0,0 +1,63 @@
+import string
+from utils import *
+
+IDEAL_Q_LENGTH = 10
+MAX_SCORE = 17
+
+def sortSecond(val):
+    return val[1]
+
+class Ranker:
+    '''
+    Holds the ranking data structure for how questions are scored and sorted
+    The q_list becomes a max priority queue, with the max score at the front.
+
+    Properties:
+        q_list: (list(Question, int)) list tuples of unranked questions and scores
+        avg_coref_len: (int) average length of coref clusters
+    '''
+    
+    def __init__(self, question_list, avg_coref_len):
+        self.q_list = [(question, 0) for question in question_list]
+        # self.doc = doc
+        self.avg_coref_len = avg_coref_len
```

Comment:
Maybe instead of calculating the avg_coref_len in ask, we could pass in the doc to here and calculate it here?

Not a big deal either way.